core/controllers/story_editor.py

```diff
@@ -0,0 +1,165 @@
+# Copyright 2018 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Controllers for the story editor."""
+from core.controllers import base
+from core.domain import acl_decorators
+from core.domain import story_domain
+from core.domain import story_services
+from core.domain import topic_domain
+from core.domain import topic_services
+import feconf
+import utils
+
+
+def _require_valid_id(entity_id):
+    """Checks whether the id received from the frontend is a valid one.
+    """
+    if not isinstance(entity_id, basestring):
+        raise base.BaseHandler.InvalidInputException(
+            Exception('All ids should be strings.'))
+
+    if len(entity_id) != 12:
+        raise base.BaseHandler.InvalidInputException(
+            Exception('The id given is invalid.'))
+
+
+class StoryEditorPage(base.BaseHandler):
+    """The editor page for a single story."""
+
+    @acl_decorators.can_edit_story
+    def get(self, topic_id, story_id):
+        """Handles GET requests."""
+
+        if not feconf.ENABLE_NEW_STRUCTURES:
+            raise self.PageNotFoundException()
+
+        _require_valid_id(story_id)
+        _require_valid_id(topic_id)
+
+        story = story_services.get_story_by_id(story_id, strict=False)
```

Comment:
Interesting, I think you'll want to fetch the topic and ensure the topic contains the story too. So that you don't serve URLs like /story_editor/randomstuffthatisnotathing/actual_story_id

Reply:
Done, also added it to skills.